src/test/java/com/hubspot/jinjava/lib/exptest/isDivisibleByExpTestTest.java

```diff
@@ -0,0 +1,78 @@
+package com.hubspot.jinjava.lib.exptest;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertEquals;
+
+import com.hubspot.jinjava.Jinjava;
+import java.util.HashMap;
+import org.junit.Before;
+import org.junit.Test;
+
+public class isDivisibleByExpTestTest {
+  private static final String DIVISIBLE_BY_TEMPLATE = "{{ %s is divisibleby %s }}";
+
+  private Jinjava jinjava;
+
+  @Before
+  public void setup() {
+    jinjava = new Jinjava();
+  }
+
+  @Test
+  public void itRequiresDividend() {
+    assertEquals(
+      jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "null", "3"), new HashMap<>()),
+      "false"
+    );
+  }
+
+  @Test
+  public void itRequiresNumericalDividend() {
+    assertEquals(
+      jinjava.render(
+        String.format(DIVISIBLE_BY_TEMPLATE, "thirty", "3"),
+        new HashMap<>()
+      ),
+      "false"
+    );
+  }
+
+  @Test
+  public void itRequiresDivisor() {
+    assertEquals(
+      jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "10", "null"), new HashMap<>()),
+      "false"
+    );
+  }
+
+  @Test
+  public void itRequiresNumericalDivisor() {
+    // Question for PR: It seems that InvalidArgumentExceptions are caught before my middleware. How to know it failed for the right reason?
+    assertEquals(
+      jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "10", "five"), new HashMap<>()),
+      "false"
+    );
+  }
+
+  @Test
+  public void itRequiresDividendIsMultipleOfDivisor() {
+    assertEquals(
+      jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "10", "5"), new HashMap<>()),
+      "true"
+    );
+
+    assertEquals(
+      jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "10", "3"), new HashMap<>()),
+      "false"
+    );
+  }
+
+  @Test
+  public void itRequiresNonZeroDivisor() {
+    assertThatThrownBy(
+        () ->
+          jinjava.render(String.format(DIVISIBLE_BY_TEMPLATE, "10", "0"), new HashMap<>())
+      )
+      .hasMessageContaining("1st argument with value 0 must be non-zero");
+  }
```

Comment:
Thanks for adding test coverage here. Can you check for the NUMBER_FORMAT exception as well?

Reply:
I don't know how to get the jinjava interpreter to take input provided as a string template into an object whose type falls outside the Number inheritance tree. That is, I don't know how to structure the test call to induce this condition.

In the process of trying I found that a template of  { 123.4 is divisibleby 3} returns "true". I would have expected a return value of false. This seems like a logic error in jinjava, though fixing it may well prove to be a breaking change, in case any client has assumed the opposite.

Reply:
not sure what you mean in the first paragraph. What have you tried?

As for the second one, I think it's worth fixing.

Reply:
I'm fixing the non-integer case to return false if it's the dividend that's fractional, and to throw an `InvalidArgumentException` if it's the divisor that's fractional. (Testing for fractional nature by comparing Math.ceil() with Math.floor() for equality).

For the first paragraph, what I mean is that all my tests invoke the system under test by passing down a string like "{10 is divisibleby 5}" or "{10 is divisibleby null}". I don't know what I might pass just before the closing "}" for a test case that violates the NUMBER_FORMAT requirement. I don't think that calling renderForResult() instead of render() in the test would buy me anything. It would appear that testing this condition requires mocking some internal jinjava interpreter class. What am I missing?

Reply:
Can you try `10 is divisibleby 'foo'`?